src/Microsoft.AspNetCore.SignalR.Client/HubConnectionBuilder.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel;
+using System.Net.Http;
+using Microsoft.AspNetCore.SignalR.Internal.Protocol;
+using Microsoft.AspNetCore.Sockets;
+using Microsoft.AspNetCore.Sockets.Client;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Microsoft.AspNetCore.SignalR.Client
+{
+    public class HubConnectionBuilder
+    {
+        private readonly Uri _url;
+        private ILoggerFactory _loggerFactory;
+        private TransportType _transportType = TransportType.All;
+        private IHubProtocol _hubProtocol;
+        private HttpMessageHandler _httpMessageHandler;
+
+        public HubConnectionBuilder(Uri url)
+        {
+            _url = url;
+        }
+
+        public HubConnectionBuilder WithLogger(ILoggerFactory loggerFactory)
```

Comment:
- I decided to go with `With...` instead of `Use` to not make the impression that this builder works in a similar way to IApplicationBuilder extension methods
- also all the `With` methods have the suffixes so that we don't have a problem in the future where we have a setting with the same type and we cannot use overloading to distinguish between options

Reply:
There's precedent for `.With`. Roslyn uses it for it's builders.